python/dataproc_templates/hive/util/hivesparkddl_to_bq.py

```diff
@@ -0,0 +1,170 @@
+            # default.emp_part
+            print("Extracting DDL for the Hive Table: "+hive_database+"."+t.name)
+            show_create = spark.sql("SHOW CREATE TABLE {}.{}".format(hive_database, t.name))
+            ddl_table = spark.sql(
+                "DESCRIBE FORMATTED {}.{}".format(hive_database, t.name))
+            show_extended=spark.sql("show table extended from `"+hive_database+"` like '"+t.name+"'")
+            ddl_table.registerTempTable("metadata")
+            show_extended.registerTempTable("show_extended")
+            info=spark.sql("select information from show_extended")
+            partition_check=show_create.filter(show_create["createtab_stmt"].contains("PARTITIONED")).head(1)
+            partition_string=""
+            if partition_check:
+                partition=show_create.first()[0].split("PARTITIONED BY (")[1].split(")")[0]
+                partitions=partition.split(",")
+                for part in partitions:
+                    partition_type=spark.sql("select data_type from metadata where col_name="+"'"+part.strip()+"'").first()[0]
+                    partition_string=partition_string+" "+part+" "+partition_type+","
+                partition_string=partition_string[:-1]
+                partition_by="PARTITIONED BY ("+partition_string+")\n"
+                first_part=show_create.first()[0].split(partitions[0])[0][:-5].split(t.name+"`")[1:]
+                first_part="".join(first_part)+")\n"
+            else:
+                partition_by=")\n"
+                first_part=show_create.first()[0].split(t.name)[1:]
+                first_part=show_create.first()[0].split(')')[0].split(t.name+"`")[1:]
+                first_part="".join(first_part)+")\n"
+            db_name=spark.sql("select data_type from metadata where col_name='Database'").first()[0]
+            table_name=spark.sql("select data_type from metadata where col_name='Table'").first()[0]
+            hdfs_file_path=info.first()[0].split("Location: ")[1].split("\n")[0]
+            format=show_create.first()[0].split("USING ")[1].split("\n")[0]
+            
+            #partition and format
+            initial="CREATE TABLE IF NOT EXISTS "+t.name
+            storage_format=format
+            print(hdfs_file_path)
+            if hive_database != 'default':
+                location_path=gcs_hdfs_staging_path+db_name+hdfs_file_path.split(db_name)[1]
+            else:
+                location_path=gcs_hdfs_staging_path+db_name+hdfs_file_path.split('warehouse')[1]
+            metadata.append([db_name,table_name,partition_string,storage_format,hdfs_file_path,location_path])
+            ddl = initial+first_part+partition_by+";\n"
+            ddls = ddls + ddl
+        rdd=spark.sparkContext.parallelize(metadata)
+        metadata_df=rdd.toDF(columns)
+        metadata_df.write.format('bigquery') .option('table', bigquery_dataset+'.'+bigquery_table) .option("temporaryGcsBucket",gcs_working_directory) .mode('append') .save()
```

Comment:
use constant already available in line 34 of template_constants file. for format("bigquery")

Reply:
Also, for option ('table') and output mode use the variable in the constant file

Reply:
done